### PR TITLE
Send Announcements for new plugins

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Screenshot from 2020-06-11 12-08-29

This should not be merged until the webhooks are setup: https://community.mattermost.com/core/pl/4ijfgw97ojfy8nqegcp1yk8f3o

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-marketplace/issues/72

### CODE REVIEW

File: .github/workflows/notify-mm-blog.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: notify-TODO
```
[the remark this answers is not in the document]
Comment (hanzei) [reply .github/workflows/notify-mm-blog.yml:1]: No. Good catch :+1:

File: .github/workflows/notify-mm-blog.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+on:
+  push:
+    branches:
+      - producion
```
[the remark this answers is not in the document]
Comment (hanzei) [reply .github/workflows/notify-mm-blog.yml:5]: Yes it should. Thanks for the catch.

File: .github/workflows/notify-mm-blog.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -x
```
[the remark this answers is not in the document]
Comment (hanzei) [reply .github/workflows/notify-mm-blog.sh:3]: TIL! Seams reasonable.

File: .github/workflows/notify-mm-blog.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+CHANGES=$(git diff HEAD~1 plugins.json)
+if [ -z "$CHANGES" ]
+then
+  echo "no changes to plugins.json"
+  echo "{}" > mattermost.json
```
[the remark this answers is not in the document]
Comment (hanzei) [reply .github/workflows/notify-mm-blog.sh:8]: Yes, it's the input for `mattermost/action-mattermost-notify`

File: .github/workflows/notify-mm-blog.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  exit 0
+fi
+
+JSON=$(git diff HEAD~1 plugins.json | grep '^+ ' | sed 's/+//g' | sed '$s/},/}/g')
+NAME=$(jq -r .manifest.name <<< "$JSON")
+VERSION=$(jq -r .manifest.version <<< "$JSON")
+MIN_SERVER_VERSION=$(jq -r .manifest.min_server_version <<< "$JSON")
+RELEASE_NOTES=$(jq -r .release_notes_url <<< "$JSON")
+
+echo '{
```
[the remark this answers is not in the document]
Comment (hanzei) [reply .github/workflows/notify-mm-blog.sh:18]: In the last few month, all changes to the Marketplace only added one plugin.  Given that, it's doesn't seam worth it to me to add the complexity to handle the case, where multiple plugin were added.

**Comment by hanzei** [comment]
@DHaussermann These changes are quite hard to QA review given that it's impossible to mock CI runs and hence a "real" commit would have to be pushed to `production`.

Also given that this PR doesn't touch the product code, would be you be find merging it without QA review?

**Comment by DHaussermann** [comment]
I agree with @hanzei it would be okay to validate this by deploying it to community. 

However one question about this... Is there a plan to allow clients to also leverage this? and if so, how will their webhooks be added? Or is this only intended to send notifications to the community server?

**Comment by hanzei** [comment]
> However one question about this... Is there a plan to allow clients to also leverage this? and if so, how will their webhooks be added? Or is this only intended to send notifications to the community server?

This PR only takes care of sending notifications to the community server. Your correctly identified the issue with opening such notifications for generic clients: How do they register? That problem is not trivial to solve and hence it got deferred for now.

**Pull request merged**